google-cloud-firestore/src/main/java/com/google/cloud/firestore/encoding/BeanMapper.java

```diff
@@ -93,18 +90,6 @@ abstract T deserialize(
       Map<TypeVariable<Class<T>>, Type> types,
       DeserializeContext context);
 
-  void ensureValidDocumentIdType(String fieldDescription, String operation, Type type) {
```

Comment:
Why move these methods from base class to a utility class?

Reply:
I am moving out all util methods that are not specific to the base class or used in both child classes.

Reply:
and this is renamed to `validateDocumentIdType` to match `validateServerTimestampType` method.